arquillian/gradle-adapter/src/main/java/org/wildfly/swarm/arquillian/adapter/gradle/GradleDependencyAdapter.java

```diff
@@ -36,187 +38,57 @@
-            // improved support for BOMs.
-            // In this particular scenario, the dependency should look like the following,
-            // org.wildfly.swarm:jaxrs-multipart -> 2018.5.0-plankton-SNAPSHOT
-
-            StringJoiner joiner = new StringJoiner(":");
-            joiner.add(coords[0]);
-
-            // Extract artifact & version details.
-            String artifactId = coords[1];
-            int idx = artifactId.indexOf(VERSION_UP);
-            if (idx != -1) {
-                String version = artifactId.substring(idx + VERSION_UP.length()).trim();
-                artifactId = artifactId.substring(0, idx).trim();
-                joiner.add(artifactId).add(version);
-                return joiner.toString();
-            }
-        }
-        // Raise an exception if we are unable to determine the dependency format.
-        throw new IllegalArgumentException("Unexpected dependency format: " + line);
-    }
-
-    private String parseLine(String line) {
-        line = line.substring(line.indexOf(PREFIX) + PREFIX.length(), line.length());
-
-        if (line.endsWith(SUFFIX)) {
-            line = line.substring(0, line.indexOf(SUFFIX));
-        }
-        return line;
     }
 
     /**
-     * Scan the project hierarchy (in case of a multi-module project) and return the project that matches the given path
-     * reference.
+     * Retrieve the project dependencies by analyzing the IdeaProject model.
      *
-     * @param project the Gradle project reference.
-     * @param path    the path to match against.
-     * @return the Gradle project reference whose path matches the given input path.
+     * @param connection the Gradle project connection.
+     * @return the project dependencies.
      */
-    private static GradleProject getProjectForDirectory(GradleProject project, Path path) {
-        if (project != null) {
-            if (project.getProjectDirectory().toPath().equals(path)) {
-                return project;
-            }
-            for (GradleProject p : project.getChildren()) {
-                GradleProject searchResult = getProjectForDirectory(p, path);
-                if (searchResult != null) {
-                    return searchResult;
-                }
-            }
-        }
-        return null;
+    private DeclaredDependencies getProjectDependencies(ProjectConnection connection) {
+        // Of the 4 available scopes for a IdeaDependency scope, we ignore "PROVIDED" & "RUNTIME".
+        final List<String> APPLICABLE_SCOPES = Arrays.asList("COMPILE", "TEST");
+        DeclaredDependencies dependencies = new DeclaredDependencies();
+
+        // 1. Get the IdeaProject model from the Gradle connection.
+        IdeaProject prj = connection.getModel(IdeaProject.class);
+
+        // 2. Find the IdeaModule that maps to the project that we are looking at.
+        Optional<? extends IdeaModule> result = prj.getModules().parallelStream()
```

Comment:
Isn't `parallelStream` a bit excessive here?

Reply:
Sorry about that. I was trying out different modes to see if I can speed up the execution time. Let me get some timing numbers and confirm if it would benefit us or not.